### DISCUSSION TITLE
Renovate not updating digests

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#15005] [Request Help] [answered]
I must be missing something glaringly obvious, but I can't get Renovate to update docker digests.

There is a `.gitlab-ci.yml` file:
```yaml
variables:
  BASE_IMAGE_NODE_14_ALPINE3_14: "node:14.19.1-alpine3.14@sha256:26af1db2431005cb34f6df34e1f2748518b8082fb4e1e6b7793de0bb95da491a"
```
For this dependency, there's a update `3f858ab7e190` according to https://hub.docker.com/_/node?tab=tags&page=1&name=14.19.1-alpine3.14

My `renovate.json5` file contains a regex manager to find them:
```json
  "regexManagers": [
    {
      "fileMatch": [
        "(^|/).gitlab-ci.yml$"
      ],
      "matchStrings": [
        "BASE_IMAGE(?:_.+)?:\\s+\"(?<depName>[^:]*):(?<currentValue>[^\\s]*)(@(?<currentDigest>sha256:[0-9a-f]*))?\""
      ],
      "datasourceTemplate": "docker",
      "versioningTemplate": "docker",
    },
  ],
```

Is there anything I need to do to persuade Renovate to update the digest?

**Comment by rarkins** [comment]
I forget if Renovate supports updating digests with regex manager?

**Reply by astellingwerf** [reply]
It is on the list of `validMatchFields` in `lib/modules/manager/regex/utils.ts`, if that means anything.

**Reply by astellingwerf** [reply]
Might it help if I switch to a regex versioning?

**Reply by viceice** [reply]
I think your regex is wrong, i know digests are working for me, checkout:
https://github.com/visualon/renovate-config/blob/bf8c1b72e7580fdec98373058a82cab157d0ddce/shared.json#L133-L144

**Comment by viceice** [comment, marked as answer]
youre regex for `currentValue` is too greedy

https://regex101.com/r/9vnywb/1

fixed version: `BASE_IMAGE(?:_.+)?:\\s+\"(?<depName>[^:]*):(?<currentValue>[^\\s@]*)(@(?<currentDigest>sha256:[0-9a-f]*))?\"`

https://regex101.com/r/2fKok9/1

**Reply by viceice** [reply]
BASE_IMAGE(?:_.+)?:\\s+\"(?<depName>[^:@]+):(?<currentValue>[^\\s@]+)(@(?<currentDigest>sha256:[0-9a-f]+))?\"

https://regex101.com/r/2fKok9/2 is event more robust, as it disallows `BASE_IMAGE_NODE_14_ALPINE3_15: "node:"` 😉

**Comment by xakraz** [comment]
Hi there!

I have a question similar to this one 😇 

## Context

We currently have repos that do not pin the digest of their docker images.

While pocing Renovate on 1 repo, I used the default `dockerfile` manager, and Renovate opened PRs to pin the digest of the images.
So far, so good.

Except that, for misc reasons, we have put the docker registry in an `ARG` variable.
So I had to set up the `regexManagers` to be able to parse the specific `FROM` statement and set the registry URL for the lookup part.

```
{
  "extends": [
    "config:base",
    ":disableDependencyDashboard",
    ":docker",
    "docker:pinDigests",
    "docker:disableMajor"
  ],

  "enabledManagers": [
    // "dockerfile",
    "regex"
  ],


  "regexManagers": [
    {
      "fileMatch": [
        "(^|/|\\.)Dockerfile$",
        "(^|/)Dockerfile\\.[^/]*$"
      ],
      "matchStrings": [
        "FROM \\$\\{DOCKER_REGISTRY\\}\/(?<repo>[a-z0-9-\/]+):(?<currentValue>[a-z0-9-]+)(@(?<currentDigest>sha256:[a-f0-9]+))?"
      ],
      "depNameTemplate": "us.gcr.io/{{{repo}}}",
      "datasourceTemplate": "docker",
      "versioningTemplate": "docker",
    }
  ],

  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "enabled": true,

      // GitHub Automation
      "addLabels": ["dependencies", "docker"],
      "automerge": true,
      "automergeType": 'pr',
      "platformAutomerge": true,
    },
    {
      "matchDatasources": ["docker"],
      "matchUpdateTypes": [
        "major"
      ],
      "enabled": false
    }
  ]
}
```

## Observations

### Note 1

* While applying this **updated config to my POC repo**, it worked as expected. Tags and digest get updated.
* However, while applying **this config on a new repo**, it doesn't pin digest and only suggests tags update.


### Note 2

If I manually pin the digest in the Dockerfile of the "new repo", the next Renovate run does what is expected: Tags and digest update through PRs.


## Question

What am I missing so that Renovate can "onboard" repos to pinned digest images without us doing it manually first ?

Thanks 🙏🏻

**Reply by viceice** [reply]
please use a new discussion and reference this.
